streams/src/main/java/org/apache/kafka/streams/kstream/internals/CogroupedStreamAggregateBuilder.java

```diff
@@ -141,6 +163,20 @@
         }
         return statefulProcessorNode;
     }
+    /**
+     * @return the new sourceName of the repartitioned source
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
it is necessary to to do an unchecked cast for the input value type. This is because cogrouped can have any type value for the group streams it intakes